src/components/ArticleDetail.jsx

```diff
@@ -90,6 +94,9 @@ function ArticleDetail() {
   return (
     <div className="article-detail-container">
       <ArticleHeader article={article} />
+
+      <ErrorMessage message={error} />
+
```

Comment:
Did you test this error handling ? 
If you search for an article_id that doesn't exist then the error message never renders as the if statement at line 91 returns loading.

Also rendering the error message as you have in the return statement wouldn't work as the ArticleHeader component relies on an article, without it the page will fail to render.

Better to use a ternary operator instead in my opinion to decide whether or not you render the page or an error.

Also both the fetchArticles and fetchComments functions fail without a valid article_id, they both set the same error state so the error message displayed would come from the comments function which you probably also didn't intend